Vision/distance_regression.py

```diff
@@ -1,99 +1,97 @@
-    img.draw_cross(160, 120, color=(30,255,10), size = 200)
-
-    # Draw blobs
     for blob in blobs:
-        # Draw a rectangle where the blob was found
-        #img.draw_circle(blob.rect(), color=(0,255,0))
-        # Draw a cross in the middle of the blob
-        img.draw_cross(blob.cx(), blob.cy(), color=(0,255,0))
-        #uart.write("Blob found at x: " + str(blob.cx()) + " y: " + str(blob.cy()) + "\n") # Sends the x and y coordinates of the blob to the serial console
-        #(Find relative cx and cy
-        rcx = blob.cx() - 160 # 50 is camera res
-        rcy = blob.cy() - 120 # 100 is camera res
-        distance_pixel = math.sqrt(rcx**2 + rcy**2)
-        angulo = math.atan2(rcy, rcx)
-        degreess = math.degrees(angulo)
-        print(f"angulo{degreess}")
-        #print(blob.cx())
-        #print(distance_pixel)
-        #print(rcx)
-        #print(rcy)
-        #print(rcx, rcy)
-        # Find distance using relative coordinates
-        #mcx = rcx * 0.2375
-        #mcy = rcy * 0.2375
-
-        #print(mcx)
-        cirle_radius=blob.enclosing_circle()[2]
-        #distance = math.sqrt(mcx**2 + mcy**2)
-        #distance=distance-circleradius
-        '''distance *= 0.1
-        #print("dist:  ",distance)
-        #print(f"distance{distance}")
-
-        # Conversion distancia relativa a distancia real
-        #---------------------------------------------------------------#
-        hcam = 5
-        hfloor = 15
-        tangente = (1/2)*(0.9*distance)*(0.045+0.45*distance**2)**(-1/2)
-        anguloTan = math.degrees(math.atan(tangente))
-        y = ((0.1+distance**2)*0.45)**(1/2)+hcam
-        anguloCentro = math.degrees(math.atan((y/distance)))
-        Y=2*(180-90-(anguloCentro-anguloTan))
-        X=(180-Y)/2
-        anguloProject =X-anguloTan
-        distTotal = (distance+(hfloor/(math.tan(math.radians(anguloProject)))))
-
-        #print(distTotal)
-
-
-
-
-        #----------------------------------------------------------------#
-
-
-    pyb.delay(50) # Pauses the execution for 50ms
-    #print(clock.fps()) # Prints the framerate to the serial console
-    #print(blob.cx(), blob.cy()) # Prints the x and y coordinates of the blob to the serial console
-    #print(blob.w()) # Prints the width of the blob to the serial console
-    #print(blob.h()) # Prints the height of the blob to the serial console
-    #uart.write(blob.w())'''
-        distTotal= 11.83*math.exp((0.0245)*distance_pixel)
-        #print(distTotal)
-    pyb.delay(50)
+        img.draw_rectangle(blob.rect(), color=(0, 255, 0))
+        img.draw_cross(blob.cx(), blob.cy(), color=(0, 255, 0))
+    return blobs
+
+def calculate_distance(blob):
+    """
+    Calculates the distance of the blob from the camera.
+    """
+    relative_cx = blob.cx() - FRAME_HEIGHT
+    relative_cy = blob.cy() - FRAME_WIDTH
+    magnitude_distance = math.sqrt(relative_cx**2 + relative_cy**2)
+    total_distance = 11.83*math.exp((0.0245)*magnitude_distance)
```

Comment:
Great refactoring!! The structure with all the functions and variable names is more readable. For this equation of the `total_distance`, I think its okay having the values obtained from the regression as numbers, as they are only being used once. 

Just add a single inline comment explaining that it is an exponential equation, and that it was calculated with real data and a regression.

Reply:
Okei great! I'll add it.